### DISCUSSION TITLE
Jenkins plugin version scheme change not caught by Renovate

### DISCUSSION THREAD

**Discussion by ckstevenson** [renovatebot/renovate#14825] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.14.1

### Please select which platform you are using if self-hosting.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

[This](https://plugins.jenkins.io/pipeline-model-definition/#releases) plugin had a recent update which Renovate didn't pick up. I assume it is because of the version scheme change? It went from `v1.9.3` to `2.2064.v5eef7d0982b_e`. It looks like the [configuration-as-code](https://plugins.jenkins.io/configuration-as-code/#releases) plugin did something similar. Jenkins UI did inform of these updates.

Here is my relevant match config:
`"matchUpdateTypes": [
               "pin",
               "digest",
               "patch",
               "minor",
               "major",
               "lockFileMaintenance"
             ]`

The debug logs didn't have anything important. Just seems to think the plugins are at the latest version. 

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Jenkins plugin dependency (repository=dcmn-com/dcanalytics-internal-services, branch=renovate/git-4.x)
       "dep": {
         "datasource": "jenkins-plugins",
         "versioning": "docker",
         "depName": "configuration-as-code",
         "currentValue": "1.55"
       }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Not a bug. Renovate can’t guess every strange versioning a package might use. Configure “versioning” with a package rule

**Reply by ckstevenson** [reply]
Okay, thanks.

**Reply by ckstevenson** [reply]
@rarkins as a test, I added the following:
`  "packageRules": [
    {
      "matchPackageNames": ["/.*/"],
      "allowedVersions": "/.*/"
    }
  ]
}`

and a value like `2.2064.v5eef7d0982b_e` is still not matched, but everything else is. Are there other ways to simply match any release string?

**Reply by rarkins** [reply]
You need to change `versioning` within a `packageRules` entry

**Reply by ckstevenson** [reply]
I missed the actual versioning page in the docs. Got it now. Thank you.
